Adds dogtag detector

benrobey opened the pull request:
# About the pull request

<!-- Remove this text and explain what the purpose of your PR is.

Mention if you have tested your changes. If you changed a map, make sure you used the mapmerge tool.
If this is an Issue Correction, you can type "Fixes Issue #169420" to link the PR to the corresponding Issue number #169420.

Remember: something that is self-evident to you might not be to others. Explain your rationale fully, even if you feel it goes without saying. -->

- Adds a new item, dogtag detector, which is similar to the data detector. It will ping bodies with dogtags that have not had the info tag removed.
- New sprite for the dogtag detector, essentially a reskinned version of the data detector in medical red.
- New (recolored) sprite for blips.

# Explain why it's good for the game

Inspired by some people posting cute screenshots in lrc of graveyards at round end.

I think it's good for the game as it doesn't affect game balance and adds a some nice flavour and enriches player stories.


# Testing Photographs and Procedure

Tested on my local by spawning USCM marines and permaing them, and making sure only the permad marines with dogtags that haven't had the info tag removed are pinged.

<details>
<summary>Screenshots & Videos</summary>


https://github.com/user-attachments/assets/c21257db-3c7e-4cc1-b2ae-f7cc9e65ddbe



</details>


# Changelog

:cl: rattybag
add: Added dogtag detector.
imageadd: added dogtag detector icons and respective blip icon.
maptweak: Alamayer map now includes dogtag detectors in lower medical storage.
/:cl:

Nanu308 reviewed (approved):
mapping is fine

DrAmazing343 commented:
What is the size of the DTD? It’s very sovl, but I’d hate to not be able to fit it into my kit as a Corpsman because it inherits the MD/DD’d three-slot enormity.

benrobey commented:
> What is the size of the DTD? It’s very sovl, but I’d hate to not be able to fit it into my kit as a Corpsman because it inherits the MD/DD’d three-slot enormity.

Good point, I think with the other detectors there are balance reasons for their sizing but for this, since it's really just an RP item I think it's fine to bump it down to `SIZE_SMALL` from `SIZE_MEDIUM`. Just pushed a commit that does this.

benrobey commented:
@Blundir Thanks for the review, I've made some changes to the sprite now, changed the shape to align with the item reduced from medium to small size and added a dogtag around the strap.

See image and video:

![image](https://github.com/user-attachments/assets/5d523bc8-178a-46ea-970a-1796c2914502)


https://github.com/user-attachments/assets/946ba7d5-fb37-4219-933b-771e606fb7ff

fira commented on code/game/objects/items/devices/dogtag_detector.dm:
```diff
@@ -0,0 +1,52 @@
+		icon_state = "[initial(icon_state)]"
+
+/obj/item/device/motiondetector/dogtag/scan()
+	set waitfor = 0
```
you don't need that

fira commented on code/game/objects/items/devices/dogtag_detector.dm:
```diff
@@ -0,0 +1,52 @@
+
+	for(var/mob/living/carbon/human/sourcemob in orange(detector_range, loc))
+		var/detected
+		if(loc == null || sourcemob == null)
```
if loc is null it's gonna be null for everyone. bail out early, not in the loop
you can also store the result of orange() and -= loc from it
then you can check z directly with `if(loc.z != sourcemob?.z) continue` and get rid of the rest
